src/sources/demo_logs.rs

```diff
@@ -156,27 +156,32 @@ async fn demo_logs_source(
         if let Some(interval) = &mut interval {
             interval.tick().await;
         }
+        emit!(&BytesReceived {
+            byte_size: 0,
+            protocol: "none",
+        });
 
         let line = format.generate_line(n);
 
         let mut stream = FramedRead::new(line.as_bytes(), decoder.clone());
         while let Some(next) = stream.next().await {
             match next {
-                Ok((events, _byte_size)) => {
+                Ok((events, byte_size)) => {
+                    let count = events.len();
+                    emit!(&EventsReceived { count, byte_size });
```

Comment:
This should also be using `BytesSizeOf::size_of` rather than the raw frame byte count.